x-pack/solutions/observability/plugins/streams_app/public/hooks/queries/use_async_sample.tsx

```diff
@@ -0,0 +1,253 @@
+import { emptyEqualsToAlways } from '../../util/condition';
+
+interface Options {
+  condition?: Condition;
+  start?: number;
+  end?: number;
+  size?: number;
+  streamDefinition: WiredStreamGetResponse;
+}
+
+export const useAsyncSample = (options: Options) => {
+  const {
+    dependencies: {
+      start: { data },
+    },
+  } = useKibana();
+
+  // Documents
+  const [isLoadingDocuments, toggleIsLoadingDocuments] = useToggle(false);
+  const [documentsError, setDocumentsError] = useState();
+  const [documents, setDocuments] = useState<Array<SearchHit<any>>>([]);
+
+  // Document counts / percentage
+  const [isLoadingDocumentCounts, toggleIsLoadingDocumentCounts] = useToggle(false);
+  const [documentCountsError, setDocumentCountsError] = useState();
+  const [approximateMatchingPercentage, setApproximateMatchingPercentage] = useState<
+    string | undefined
+  >();
+
+  const [refreshId, setRefreshId] = useState(0);
+
+  const convertedCondition = useMemo(() => {
+    return options.condition ? emptyEqualsToAlways(options.condition) : undefined;
+  }, [options.condition]);
+
+  useEffect(() => {
+    if (!convertedCondition || !options.start || !options.end) {
+      setDocuments([]);
+      setApproximateMatchingPercentage(undefined);
+      return;
+    }
+
+    const runtimeMappings = getRuntimeMappings(options.streamDefinition, convertedCondition);
+
+    // Documents
+    toggleIsLoadingDocuments(true);
+    setDocuments([]);
+    const documentSubscription = data.search
+      .search({
+        params: {
+          index: options.streamDefinition.stream.name,
+          body: getDocumentsSearchBody(convertedCondition, options, runtimeMappings),
+        },
+      })
+      .subscribe({
+        next: (result) => {
+          if (result.rawResponse.hits?.hits) {
+            setDocuments((prev) =>
+              prev.concat(result.rawResponse.hits.hits.map((hit) => hit._source))
+            );
+          }
+          if (!result.isRunning) {
+            toggleIsLoadingDocuments(false);
+          }
+        },
+        error: (e) => {
+          setDocumentsError(e);
+          toggleIsLoadingDocuments(false);
+        },
+      });
+
+    // Document counts
+    toggleIsLoadingDocumentCounts(true);
+    setApproximateMatchingPercentage(undefined);
+    const documentCountsSubscription = data.search
+      .search({
+        params: {
+          index: options.streamDefinition.stream.name,
+          body: getDocumentCountsSearchBody(convertedCondition, options, runtimeMappings),
+        },
+      })
+      .subscribe({
+        next: (result) => {
+          if (!result.isRunning) {
+            toggleIsLoadingDocumentCounts(false);
+          }
+          // Aggregations don't return partial results so we just wait until the end
+          if (!result.isRunning && result.rawResponse?.aggregations) {
+            const randomSampleDocCount = result.rawResponse.aggregations.sample.doc_count;
```

Comment:
we need to divide this by the sampling factor to get an accurate result. 
The docs are not clear on this, but the same behaviour here applies: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-random-sampler-aggregation.html#random-sampler-special-cases

Reply:
Ah! Thanks @LucaWintergerst 👌